evaluate_library_access_from_output_tsv/read_and_merge_library_access_datasets.R

```diff
@@ -0,0 +1,43 @@
+#### Read the datasets ####
```

Comment:
I don't think this should be a separate file. Rather integrate it into the script. First, with the tidyverse the code will become much shorter. Second, it's not a generalized operation (reading this file & merging it in a specific way). If you move this functionality to the script that uses and keep it in a function, then that is probably okay (but likely unnecessary). It's the splitting this functionality into a new file and the namespace tactics that are most problematic:

> '<<-' at several places below will add the variable into our global scope, beyond this function.

This makes the code hard to follow and can lead to certain bugs.

With the tidyverse, it's often most preferred to pipe lot's of commands together. For example, this entire function could probably become one chain.

Reply:
This same code is used in both `evaluate_library_access_from_output_tsv.Rmd` and `create_stratefied_sample_of_dois.R`, and so seemed justified to me to split into a function. I agree with your point about using `<<-`, and normally wouldn't do it, but that also seems justified to me in this case (in the context of wrapping this in a function), as, in line with your comment, this is not a generalized operation (even if you were noting that to argue against wrapping it in a function).

I assumed that you would globally argue against repeating code across files, and so took this function approach; so that we're on the same page, before I make this change, are you ok with repeating the read-and-merge across files in this specific case?

Reply:
> are you ok with repeating the read-and-merge across files in this specific case?

Yes that's okay to have some repetition of parts of data processing code. The tidyverse APIs are already very high level. If you find yourself constantly repeating the same read-and-merge operations, then maybe you should save those results as a TSV (not saying that's the case here).

Reply:
Implemented in aac5bf5ef.